enterprise/bloom-index/src/main/java/org/neo4j/kernel/api/impl/bloom/PartitionedInsightBloomWriter.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.impl.bloom;
+
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.Query;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.neo4j.kernel.api.impl.index.partition.AbstractIndexPartition;
+
+/**
+ * Schema Lucene index writer implementation that supports writing into multiple partitions and creates partitions
+ * on-demand if needed.
+ * <p>
+ * Writer threats partition as writable if partition has number of live and deleted documents that is less then configured
+ * {@link #MAXIMUM_PARTITION_SIZE}.
+ * First observable partition that satisfy writer criteria is used for writing.
+ */
+class PartitionedInsightBloomWriter
```

Comment:
this writer is almost a copy of PartitionedIndexWriter, why do we need to create it why not reuse it for this indexes as well?